src/modm/platform/i2c/rp/i2c_master.cpp.in

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022, Andrey Kunitsyn
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include "i2c_master_{{ id }}.hpp"
+#include <modm/platform/core/resets.hpp>
+
+
+namespace
+{
+	struct SimpleWait
+	{
+		using State = uint_fast32_t;
+		static State start() {
+			return 100'000;
+		}
+		static bool check(State& s) {
+			return --s == 0;
+		}
+	};
+}
+// ----------------------------------------------------------------------------
+
+void modm::platform::I2cMaster{{ id }}::hwReset()
+{
+	Resets::reset(RESETS_RESET_I2C{{ id }}_BITS);
+}
+
+void modm::platform::I2cMaster{{ id }}::hwUnReset()
+{
+	Resets::unresetWait(RESETS_RESET_I2C{{ id }}_BITS);
+}
+
+void
+modm::platform::I2cMaster{{ id }}::reset()
+{
+	errorState = Error::SoftwareReset;
+	restartOnNext = false;
+}
+
+bool
+modm::platform::I2cMaster{{ id }}::start(I2cTransaction *transaction, ConfigurationHandler handler)
+{
+	if (!transaction)
+	{
+		return true;
+	}
+	if (not transaction->attaching())
+	{
+		transaction->detaching(modm::I2c::DetachCause::FailedToAttach);
+		// return false; // done at the end of the function
```

Comment:
Why not just return here?

Reply:
just source based on more complex implementation from another platforms